helper.go

```diff
@@ -277,6 +271,134 @@ func arrayToKV(m map[string]RedisMessage, arr []RedisMessage, keys []string) map
 	return m
 }
 
+func clusterMGet(client Client, ctx context.Context, keys []string) (ret map[string]RedisMessage, err error) {
+	ret = make(map[string]RedisMessage, len(keys))
+	if len(keys) == 0 {
+		return ret, nil
+	}
+
+	// Map slot -> index in cmds.s
+	slotIdx := make(map[uint16]int, len(keys)/2)
+	cmds := mgetcmdsp.Get(0, len(keys))
+	defer mgetcmdsp.Put(cmds)
+
+	for _, key := range keys {
+		s := slot(key)
+		idx, ok := slotIdx[s]
+		if !ok {
+			// first key in this slot: create a new MGET
+			idx = len(cmds.s)
+			slotIdx[s] = idx
+			cmds.s = append(cmds.s, client.B().Mget().Key(key).Build().Pin())
+			continue
+		}
+
+		c := cmds.s[idx]
+		args := c.Commands()                 // ["MGET", k1, k2, ...]
+		mk := client.B().Mget().Key(args[1]) // first existing key
+		for i := 2; i < len(args); i++ {     // remaining existing keys
+			mk = mk.Key(args[i])
+		}
+		mk = mk.Key(key) // add the new key
+
+		intl.PutCompletedForce(c)
+		cmds.s[idx] = mk.Build().Pin()
+	}
+
+	resps := client.DoMulti(ctx, cmds.s...)
+	defer resultsp.Put(&redisresults{s: resps})
+
+	// Convert each response to an array once
+	values := make([][]RedisMessage, len(cmds.s))
```

Comment:
only using:
1. slotIdx map[uint16]int for slot → command index, and
2. the pooled cmds.s slice for the actual Completed commands.